src/traversals/diffusion.jl

```diff
@@ -0,0 +1,97 @@
+"""
+    diffusion(g, p, num_steps)
+
+Run diffusion simulation on `g` for `num_steps` steps with spread
+probabilities based on `p`. Return a vector with the set of vertices
+reached at each step of the simulation.
+
+### Optional Arguments
+- `initial_infections=sample(vertices(g), 1)`: A list of vertices that
+are infected at the start of the simulation.
+- `watch=Set()`: While simulation is always run on the full graph,
+specifying `watch` limits reporting to a specific set of vertices reached
+during the simulation. If left as an empty set, all vertices will be watched.
+- `normalize=false`: if `false`, set the probability of spread from a vertex ``i`` to
+each of the out_neighbors of ``i`` to ``p``. If `true`, set the probability of spread
+from a vertex ``i`` to each of the `out_neighbors` of ``i`` to
+``\\frac{p}{outdegreee(g, i)}``.
+"""
+
+function diffusion(g::AbstractGraph,
+                   p::AbstractFloat,
+                   num_steps::Integer;
+                   watch::Set=Set(),
+                   initial_infections::Set=Set(LightGraphs::sample(vertices(g), 1)),
```

Comment:
Don't do that. You don't want the user to have to create a Set object before calling this. He should be able to pass in a vector of vertices and have it do the right thing. (We don't require this of any other LG function that takes a subset of vertices as a parameter.)